compiler/noirc_frontend/src/tests.rs

```diff
@@ -1461,13 +1461,15 @@ fn struct_numeric_generic() {
         inner: u64
     }
 
-    fn bar<let N: Foo>() { }
+    struct Bar<let N: Foo> { }
```

Comment:
Why was this changed from a function to a struct?

Reply:
We still resolve function generics in `add_generics` when we are defining a function meta and I wanted to trigger a def collector error rather than a resolver error. 

I have made two separate tests now, one that checks a function and one that checks a struct.